presto-orc/src/main/java/com/facebook/presto/orc/writer/DictionaryBuilder.java

```diff
@@ -81,42 +70,47 @@ public long getRetainedSizeInBytes()
         return INSTANCE_SIZE + elementBlock.getRetainedSizeInBytes() + blockPositionByHash.sizeOf();
     }
 
-    public Block getElementBlock()
+    public int compareIndex(int left, int right)
     {
-        return elementBlock;
+        return elementBlock.compareTo(
+                left,
+                0,
+                elementBlock.getSliceLength(left),
+                elementBlock,
+                right,
+                0,
+                elementBlock.getSliceLength(right));
```

Comment:
Same comment as the one for the last PR: it is fairly hard to follow all the refactor + new changes within one commit. Let's maybe start with this PR to break it into multiple commits. Some commits should only contain code movement. Some commits should contain interface changes. Some commits have logic changes. Data structure change like BigIntArray to int[] should also be a separate commit.

Reply:
I will break them into smaller commits and update this PR with a list of commits for easy reviewing.